### PR TITLE
Remove BoringSSL

### PR DESCRIPTION
**Pull request by 0xTim**
Spike removing BoringSSL now that Swift Crypto has RSA support

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -14,13 +14,14 @@ let package = Package(
         MANGLE_END */
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-crypto.git", from: "1.0.0")
```
Comment (shengchalover) [inline Package.swift:17]: omg

File: Package.swift [hunk 2]
```diff
@@ -14,13 +14,14 @@ let package = Package(
...
+        .package(url: "https://github.com/apple/swift-crypto.git", .branch("main"))
```
Comment (tonyarnold) [inline Package.swift:17]:
```suggestion
        .package(url: "https://github.com/apple/swift-crypto.git", "2.0.0" ..< "3.0.0")
```

**Comment by chhetri-rahul** [comment]
@tonyarnold @shengchalover this pr has been open for a while, are there plans to merge it?

**Comment by shengchalover** [comment]
@chhetri-rahul hey, I am mostly a lurker, cannot comment on this.

**Comment by 0xTim** [comment]
@chhetri-rahul we need to port over the APIs to use Swift Crypto and investigate if they provide everything we need

**Comment by 0xTim** [comment]
Closing in favour of #97

**Pull request closed without merging**